idx/cassandra/cassandra.go

```diff
@@ -343,12 +343,23 @@ func (c *CasIdx) updateCassandra(now uint32, inMemory bool, archive idx.Archive,
 func (c *CasIdx) rebuildIndex() {
 	log.Info("cassandra-idx: Rebuilding Memory Index from metricDefinitions in Cassandra")
 	pre := time.Now()
-	var defs []schema.MetricDefinition
+	gate := make(chan struct{}, c.cfg.initLoadConcurrency)
+	var wg sync.WaitGroup
 	var num int
 	for _, partition := range cluster.Manager.GetPartitions() {
-		defs = c.LoadPartitions([]int32{partition}, defs[:0], pre)
-		num += c.MemoryIndex.LoadPartition(partition, defs)
-	}
+		wg.Add(1)
+		go func(p int32) {
+			gate <- struct{}{}
+			defer func() {
+				wg.Done()
+				<-gate
+			}()
+			var defs []schema.MetricDefinition
```

Comment:
We should try and reuse these slices where possible. If concurrency is set to 1, then we can re-use the same slice for every partition and reduce allocations.

Perhaps a sync.Pool could be used.